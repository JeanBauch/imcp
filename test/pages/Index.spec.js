import { mount } from '@vue/test-utils';

import Index from '~/pages/index.vue';

describe('Pages:index', () => {
	let mountConfig;

	beforeEach(() => {
		mountConfig = {
			stubs: {
				ButtonInstrument: true,
				Header: true,
				Banner: true,
				CardEventos: true,
				Button: true,
				CardNoticia: true,
				CardMusicPlayer: true,
				Footer: true,
			},
		};
	});

	describe('Snapshots', () => {
		it('Should render all child componets', () => {
			const wrapper = mount(Index, mountConfig);
			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
