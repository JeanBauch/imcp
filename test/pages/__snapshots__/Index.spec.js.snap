// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages:index Snapshots Should render all child componets 1`] = `
"<div class=\\"bg-background flex flex-col gap-4 md:gap-8\\">
  <section class=\\"h-screen bg-background md:bg-black relative\\">
    <banner-stub></banner-stub>
  </section>
  <section class=\\"container mx-auto\\">
    <div class=\\"flex flex-col-reverse lg:flex-row lg:justify-between px-6 sm:px-9 lg:px-24 gap-8 lg:gap-0\\">
      <div class=\\"flex flex-col items-start justify-center w-full xl:w-4/5 gap-4\\">
        <h3 class=\\"text-black text-3xl 2xl:text-4xl w-full sm:w-5/6 font-semibold\\">
          Instituto dos <span class=\\"text-primary-bordo-500\\">Meninos Cantores de Petropólis</span></h3>
        <p class=\\"text-base\\">
          O Instituto dos Meninos Cantores de Petrópolis administra a educação de crianças e jovens, dos 8 aos 18 anos de idade, através do Curso de Aprendizes e dos corais Canarinhos de Petrópolis e Meninas dos Canarinhos de Petrópolis.
        </p>
        <button-stub title=\\"Saiba Mais\\" mediumsize=\\"true\\" color=\\"bordo\\" class=\\"mt-4\\"></button-stub>
      </div>
      <div class=\\"flex w-full lg:justify-end\\">
        <div class=\\"w-80 xl:w-4/5 h-60 lg:h-60 xl:h-96 2xl:h-[460px] bg-cover bg-center bg-[url('https://www.diariodepetropolis.com.br/Conteudo/Dados_DRPTR14/acervo/1/202010/fotos/0640_x_0427_20201028181244_XMBOJ.jpg')]\\" style=\\"border-radius: 60% 40% 33% 67% / 36% 51% 49% 64%;\\"></div>
      </div>
    </div>
  </section>
  <section class=\\"container mx-auto my-10\\">
    <div class=\\"h-full flex flex-col gap-4 pl-6 sm:pl-9 lg:px-24\\">
      <h3 class=\\"text-3xl 2xl:text-4xl font-semibold\\">
        Escola de <span class=\\"text-primary-orange-500\\">Música</span></h3>
      <ul role=\\"list\\" class=\\"block md:hidden marker:text-primary-orange-500 list-disc pl-5 space-y-2 font-light\\">
        <li>Aulas de instrumentos musicais</li>
        <li>Professores renomados</li>
        <li>Teoria musical, solfejo e percepção musical</li>
        <li>Valores franciscanos</li>
      </ul>
      <div class=\\"snap-x flex snap-mandatory overflow-scroll h-52 md:h-full md:overflow-hidden md:snap-none md:grid md:grid-cols-3 md:gap-6 mt-4\\">
        <buttoninstrument-stub title=\\"Aprendizes\\" icon=\\"passarinho\\" color=\\"yellow\\" class=\\"snap-center flex-shrink-0 w-5/12 mr-3 md:w-full md:max-h-72 2xl:max-h-fit md:mr-0\\"></buttoninstrument-stub>
        <buttoninstrument-stub title=\\"piano\\" icon=\\"piano\\" color=\\"orange\\" class=\\"snap-center flex-shrink-0 w-5/12 mr-3 md:w-full md:max-h-72 2xl:max-h-fit md:mr-0\\"></buttoninstrument-stub>
        <buttoninstrument-stub title=\\"regência\\" icon=\\"regencia\\" color=\\"bordo\\" class=\\"snap-center flex-shrink-0 w-5/12 mr-3 md:w-full md:max-h-72 2xl:max-h-fit md:mr-0\\"></buttoninstrument-stub>
        <buttoninstrument-stub title=\\"violino\\" icon=\\"violino\\" color=\\"beige\\" class=\\"snap-center flex-shrink-0 w-5/12 mr-3 md:w-full md:max-h-72 2xl:max-h-fit md:mr-0\\"></buttoninstrument-stub>
        <buttoninstrument-stub title=\\"Violão\\" icon=\\"violao\\" color=\\"blue\\" class=\\"snap-center flex-shrink-0 w-5/12 mr-3 md:w-full md:max-h-72 2xl:max-h-fit md:mr-0\\"></buttoninstrument-stub>
        <buttoninstrument-stub title=\\"Órgão\\" icon=\\"orgao\\" color=\\"yellow\\" class=\\"snap-center flex-shrink-0 w-5/12 mr-3 md:w-full md:max-h-72 2xl:max-h-fit md:mr-0\\"></buttoninstrument-stub>
      </div>
    </div>
  </section>
  <section class=\\"w-full bg-primary-beige-300 pt-10\\">
    <div class=\\"container mx-auto px-6 sm:px-9 lg:px-24\\">
      <div class=\\"w-full mb-6\\">
        <h3 class=\\"text-2xl 2xl:text-3xl font-light\\">
          AGENDA
        </h3>
        <h3 class=\\"text-3xl 2xl:text-4xl font-semibold\\">
          Próximos <span class=\\"text-primary-bordo-500\\">eventos.</span></h3>
      </div>
      <main class=\\"flex flex-col md:flex-row gap-4 lg:gap-8\\">
        <cardeventos-stub color=\\"bordo\\"></cardeventos-stub>
        <cardeventos-stub color=\\"blue\\"></cardeventos-stub>
      </main>
      <div class=\\"flex justify-center mt-6 lg:mt-8\\">
        <button-stub title=\\"Ver agenda completa\\" mediumsize=\\"true\\" color=\\"beige\\" class=\\"hover:motion-safe:animate-bounce\\"></button-stub>
      </div>
    </div>
    <div class=\\"w-full h-20 -mt-4 bg-primary-beige-500 bg-opacity-60 clip-path-waves-mobile rotate-180\\"></div>
  </section>
  <section class=\\"container mx-auto mt-10\\">
    <div class=\\"flex flex-col gap-8 px-6 sm:px-9 lg:px-24\\">
      <div>
        <h3 class=\\"text-3xl 2xl:text-4xl font-semibold\\">
          Portal de <span class=\\"text-secondary-blue-500\\">Notícias.</span></h3>
      </div>
      <main class=\\"grid grid-cols-1 sm:grid-cols-2 xl:grid-cols-3 gap-4 relative\\">
        <cardnoticia-stub></cardnoticia-stub>
        <cardnoticia-stub></cardnoticia-stub>
        <cardnoticia-stub></cardnoticia-stub>
      </main>
      <div class=\\"flex justify-center lg:mt-5\\">
        <button-stub title=\\"Ver tudo\\" mediumsize=\\"true\\" color=\\"yellow\\"></button-stub>
      </div>
    </div>
  </section>
  <section class=\\"container mx-auto my-6 md:mt-2\\">
    <div class=\\"flex flex-col gap-5 md:gap-8 px-6 sm:px-9 lg:px-24\\">
      <div class=\\"flex justify-center sm:justify-start\\">
        <h3 class=\\"text-3xl 2xl:text-4xl font-semibold\\">
          Conheça nossas <span class=\\"text-primary-bordo-500\\">músicas.</span></h3>
      </div>
      <cardmusicplayer-stub></cardmusicplayer-stub>
    </div>
  </section>
</div>"
`;
